### PR TITLE
update logic + config management refactor

### PR DESCRIPTION
**Pull request by tarunKoyalwar**
### Proposed Changes

- replace old version check endpoint with new pdtm version check endpoint
- uses standard/default tool self update logic from `utils/update` + shows release notes on update
- config management refactor (Ref: https://github.com/projectdiscovery/nuclei/issues/3572)
- update logic refactor 
   - template update/ install logic is done from `internal.installer` package which uses most of existing logic from utils/update
   - centrailized template management with `templateManager`
   - ref: https://github.com/projectdiscovery/nuclei/issues/3428
```
// package install provides helper functions for all download and installation of following tasks:
// 1. downloading and install nuclei templates
// 2. download and update nuclei binary (i.e self update)
// 3. version check for nuclei binary & templates
```
- based on https://github.com/projectdiscovery/utils/pull/131
  - Nuclei now properly handles update logic if nuclei is installed from dev branch 
  >  If installed tool is from dev branch ex: v2.9.1-dev it is flaged as outdated and available for update if v2.9.1 is released

### CODE REVIEW

**Comment by tarunKoyalwar** [comment]
- all config related stuff is available at v2/pkg/catalog/config
- all template management stuff is available at v2/internal/installer

**Comment by tarunKoyalwar** [comment]
Screenshot 2023-04-19 at 9 11 36 PM
Screenshot 2023-04-19 at 9 11 46 PM

**Review by ehsandeep — APPROVED** [review]

File: v2/cmd/nuclei/main.go [hunk 1]
```diff
@@ -342,62 +336,97 @@ on extensive configurability, massive extensibility and ease of use.`)
...
+	if cfg.CustomGithubTemplatesDirectory != "" {
+		gologger.Info().Msgf("Custom Github templates location: %s ", cfg.CustomGithubTemplatesDirectory)
+	}
+	os.Exit(0)
+}
+
 func init() {
 	// print stacktrace of errors in debug mode
-	if os.Getenv("DEBUG") != "" {
+	if os.Getenv("DEBUG") == "True" || os.Getenv("DEBUG") == "true" {
```
Comment (Mzack9999) [inline v2/cmd/nuclei/main.go:429]: what about `strings.equalfold`?

**Review by Mzack9999 — APPROVED** [review]
lgtm - suggesting minor change for string comparison

File: v2/cmd/nuclei/main.go (hunk 1 above)
Comment (tarunKoyalwar) [reply v2/cmd/nuclei/main.go]: done

**Pull request merged**